### ISSUE TITLE
Add COSMIC theme

### ISSUE THREAD

**Issue by dacid44** [ozwaldorf/carburetor#6]
Would you be interested in a PR for a Carburetor theme for COSMIC DE? I'm working on building it right now, but I thought I'd ask since Cosmic is still pretty unstable, and so it might require some extra effort to keep it up to date.

**Comment by dacid44** [comment]
Yeah, what I'm currently working on is building a whiskers template file for a cosmic theme. I've actually just finished that, and once I've tested it out, I'll probably submit it to both repos. The challenge with nix will be deciding where to put it, since COSMIC doesn't really have an "available themes" location. It has a standardized format for importing and exporting themes, but when you import a theme file, it breaks it up into pieces and (I think) applies some color modifications that aren't easy to replicate before saving it to its config.

**Comment by ozwaldorf** [comment]
That sounds great! For now then we can just have a build in nix using mkWhiskersDerivation on your repo, and if merged to catppuccin, we can switch it to upstream

As far as applying if that's the case, then just nix as a builder is fine and we'll commit the files like the other configs

**Comment by dacid44** [comment]
Ok, I did some research and it looks like there is a way to programmatically apply a cosmic theme, but I don't think there are any existing programs that do it. I could possibly write a small Rust program to use `libcosmic` to programmatically apply the theme, but do you think it's worth the extra complexity over just leaving the theme at a predetermined place in the user's home directory?

Also, I think my PR to the catppuccin/cosmic-desktop repo will probably be merged soon. https://github.com/catppuccin/cosmic-desktop/pull/8

**Comment by ozwaldorf** [comment]
> Ok, I did some research and it looks like there is a way to programmatically apply a cosmic theme, but I don't think there are any existing programs that do it. I could possibly write a small Rust program to use libcosmic to programmatically apply the theme, but do you think it's worth the extra complexity over just leaving the theme at a predetermined place in the user's home directory?

nice work thanks for looking into that! honestly it's up to you, I'm happy with just moving them to a common place for now, or even leaving it completely up to the user to install, either from the raw derivation, or from provided precompiled ones in the src directory. If there's a good way in the future to install the themes, we can add it
